File: frontend/assets/js/log-level.js
class LogLevel {

  /**
   * Constructor!
   * @param {null|{ext:string, key: string, name: string}} obj
   */
  constructor(obj = null) {

    /**
     * Extension of file
     * @type {string}
     */
    this.ext = '';

    /**
     * Key of level log
     * @type {string}
     */
    this.key = '';

    /**
     * Name of level log
     * @type {string}
     */
    this.name = '';

    if (obj) {
      this.init(obj);
    }

  }

  /**
   * Init by object
   * @param {{ext:string, key: string, name: string}} obj
   */
  init(obj) {

    if (obj.ext) {
      this.ext = obj.ext;
    }

    if (obj.name) {
      this.name = obj.name;
    }

    if (obj.key) {
      this.key = obj.key;
    }

  }

}

export default LogLevel;

File: backend/Common/LogLevel.cpp
#include "LogLevel.h"

namespace Common {

    LogLevel::LogLevel() {}

    void from_json(const nlohmann::json &j, LogLevel &level) {

        level.key = j.at("key").get<std::string>();
        level.ext = j.at("ext").get<std::string>();
        level.name = j.at("name").get<std::string>();

    }

    void to_json(nlohmann::json &j, const LogLevel &level) {

        j = nlohmann::json{{"key",  level.key},
                           {"name", level.name},
                           {"ext",  level.ext}};
    }


}